File: java/Inheritance Case Study/NormalParcel.java
class NormalParcel extends Parcel {
	public void displayNormalReceipt(){
		int overWeightCharges = 0;
 		int totalOverWeight = 0;
		int totalCharges = this.fee;
		if(this.weight>900 ){
			totalOverWeight = this.weight - 900 ;
			overWeightCharges =  totalOverWeight * this.feePerGram ;
			totalCharges =  totalCharges + (this.feePerGram * totalOverWeight );
		} 
		System.out.println("\n\nShipment Receipt");
		System.out.println("-------------------------");
		System.out.println("Receipt No. : "+this.id);
		System.out.println("Sender Name : "+this.senderName);
		System.out.println("Sender Address : "+this.senderAddress);
		System.out.println("Receiver Name : "+this.receiverName);
		System.out.println("Receiver Address : "+this.receiverAddress);
		System.out.println("Parcel Weight : "+this.weight);
		if(totalOverWeight>0){
			System.out.println("OverWeight : "+totalOverWeight+"g");
		}else{
			System.out.println("OverWeight : No");
		}
		System.out.println("Basic Charges : "+this.fee);
		System.out.println("OverWeight Charges: "+overWeightCharges);
		System.out.println("Shipment Total Charges: "+totalCharges);
		System.out.println("Type of Shipment: Normal");
	}
}
 

File: cpp/Video 04 Code - Inheritance/CS/Encapsulation/Encapsulation/NormalParcel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Encapsulation
{
    public class NormalParcel : Parcel
    {
        
        public void DisplayNormalReceipt()
        {
            int overWeightCharges = 0;
          
            int totalOverWeight = 0;
            int totalCharges = this.fee;
            if (this.weight > 900)
            {
                totalOverWeight = this.weight - 900;
                overWeightCharges = totalOverWeight * this.feePerGram;
                totalCharges = totalCharges + (this.feePerGram * totalOverWeight);
            }
            Console.WriteLine("Shipment Receipt");
            Console.WriteLine("-------------------------");
            Console.WriteLine("Receipt No. : " + this.id);
            Console.WriteLine("Sender Name : " + this.senderName);
            Console.WriteLine("Sender Address : " + this.senderAddress);
            Console.WriteLine("Receiver Name : " + this.receiverName);
            Console.WriteLine("Receiver Address : " + this.receiverAddress);
            Console.WriteLine("Parcel Weight : " + this.weight);
            if (totalOverWeight > 0)
            {
                Console.WriteLine("OverWeight : " + totalOverWeight + "g");
            }
            else
            {
                Console.WriteLine("OverWeight : No");
            }
            Console.WriteLine("Basic Charges : " + this.fee);
            Console.WriteLine("OverWeight Charges: " + overWeightCharges);
            Console.WriteLine("Shipment Total Charges: " + totalCharges);
            Console.WriteLine("Type of Shipment: Normal");
        }
    }
}
